### PR TITLE
[Key Vault] Python 3-style typing and formatting

### PR DESCRIPTION
**Pull request by mccoyp**
# Description

This updates KV packages to use Python 3-style type hints everywhere, typing for tools like `mypy`, and formatting such as f-strings and docstrings.

This also syncs the `_shared` folders between Certs, Keys, and Secrets, which includes sharing the `send_request` method implementation between all their clients (this method was initially just in `azure-keyvault-keys`).

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py [hunk 1]
```diff
@@ -105,13 +112,15 @@ def get_role_assignment(self, scope, name, **kwargs):
         return KeyVaultRoleAssignment._from_generated(assignment)
 
     @distributed_trace
-    def list_role_assignments(self, scope, **kwargs):
-        # type: (Union[str, KeyVaultRoleScope], **Any) -> ItemPaged[KeyVaultRoleAssignment]
+    def list_role_assignments(
+        self, scope: "Union[str, KeyVaultRoleScope]", **kwargs: "Any"
```
Comment (YalinLi0312) [inline sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py:116]: You can remove hints for kwargs which is the comment I got: https://github.com/Azure/azure-sdk-for-python/pull/28095#discussion_r1060868828

Comment (mccoyp) [reply sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py:116]: That's good to know, thanks for the heads-up! I'll push an update in a sec

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py [hunk 2]
```diff
@@ -38,8 +38,9 @@ class KeyVaultAccessControlClient(KeyVaultClientBase):
     # pylint:disable=protected-access
 
     @distributed_trace
-    def create_role_assignment(self, scope, definition_id, principal_id, **kwargs):
-        # type: (Union[str, KeyVaultRoleScope], str, str, **Any) -> KeyVaultRoleAssignment
+    def create_role_assignment(
+        self, scope: "Union[str, KeyVaultRoleScope]", definition_id: str, principal_id: str, **kwargs: "Any"
```
Comment (YalinLi0312) [inline sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py:42]: nit: Why not get rid of `TYPE_CHECKING`? Then don't need the quotes for all type hints.

Comment (YalinLi0312) [reply sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py:42]: You can remove hints for `kwargs` which is the comment I got: https://github.com/Azure/azure-sdk-for-python/pull/28095#discussion_r1060868828

Comment (mccoyp) [reply sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py]: > nit: Why not get rid of TYPE_CHECKING?

That's something I'd like to do eventually, but given that there are already a lot of changes in this PR, I was thinking of saving that for another time.

**Review by YalinLi0312 — APPROVED** [review]
LGTM!

**Pull request merged**